### ISSUE TITLE
Using your test code, why did I only get 0.275(EAO) on VOT2018

### ISSUE THREAD

**Issue by ZZXin** [zhanglichao/updatenet#27]
Hi，
Thanks for your awesome job.
Now,when I test your code on VOT2018,But I only get 0.275(EAO), Is there something wrong with my configuration?The following is my test code, imitating pysot's test.py to make some changes:

from __future__ import absolute_import
from __future__ import division
from __future__ import print_function
from __future__ import unicode_literals

import argparse
import os
import random
import cv2
import torch
import numpy as np

import vot
import sys
from os.path import realpath, dirname, join

from net_upd import SiamRPNBIG
from updatenet import UpdateResNet
from run_SiamRPN_upd import SiamRPN_init, SiamRPN_track_upd
from utils import get_axis_aligned_bbox, cxy_wh_2_rect

from datasets import DatasetFactory
from utils0.region import vot_overlap, vot_float2str

parser = argparse.ArgumentParser(description='siamrpn tracking')
parser.add_argument('--dataset', default='VOT2018', type=str,
        help='datasets')
parser.add_argument('--snapshot', default='', type=str,
        help='snapshot of models to eval')
parser.add_argument('--video', default='', type=str,
        help='eval one special video')
parser.add_argument('--seed', default=-1, type=int,
        help='random seed')
parser.add_argument('--vis', action='store_true',
        help='whether visualzie result')
parser.add_argument('--skip', action='store_true',
        help='skip tested sequences')
args = parser.parse_args()

torch.set_num_threads(1)

def seed_torch(seed=0):
    random.seed(seed)
    os.environ['PYTHONHASHSEED'] = str(seed)
    np.random.seed(seed)
    torch.manual_seed(seed)
    torch.cuda.manual_seed(seed)
    torch.backends.cudnn.deterministic = True
    torch.backends.cudnn.benchmark = True

def main():

    cur_dir = os.path.dirname(os.path.realpath(__file__))
    #dataset_root = os.path.join(cur_dir, '../testing_dataset', args.dataset)
    dataset_root = os.path.join('/media/db/dbVolume/Datasets/pysot', args.dataset)

    # load net
    net_file = join(realpath(dirname(__file__)), '/home/db/Subject/PyCharm_Hub/updatenet/templates_generate/SiamRPNBIG.model')
    net = SiamRPNBIG()
    net.load_state_dict(torch.load(net_file))
    net.eval().cuda()
    updatenet = UpdateResNet()
    update_model = torch.load('../models/vot2018.pth.tar')['state_dict']

    updatenet.load_state_dict(update_model)
    updatenet.eval().cuda()

    # create dataset
    dataset = DatasetFactory.create_dataset(name=args.dataset,
                                            dataset_root=dataset_root,
                                            load_img=False)

    model_name = args.snapshot.split('/')[-1].split('.')[0]
    total_lost = 0
    if args.dataset in ['VOT2016', 'VOT2018', 'VOT2019']:
        # restart tracking
        for v_idx, video in enumerate(dataset):
            if args.video != '':
                # test one special video
                if video.name != args.video:
                    continue
            if args.skip and os.path.isdir(os.path.join('results', args.dataset, model_name,
                'baseline', video.name)):
                print('{} skip {}'.format(v_idx+1, video.name))
                continue
            frame_counter = 0
            lost_number = 0
            toc = 0
            pred_bboxes = []
            for idx, (img, gt_bbox) in enumerate(video):
                if len(gt_bbox) == 4:
                    gt_bbox = [gt_bbox[0], gt_bbox[1],
                       gt_bbox[0], gt_bbox[1]+gt_bbox[3]-1,
                       gt_bbox[0]+gt_bbox[2]-1, gt_bbox[1]+gt_bbox[3]-1,
                       gt_bbox[0]+gt_bbox[2]-1, gt_bbox[1]]
                tic = cv2.getTickCount()
                if idx == frame_counter:
                    cx, cy, w, h = get_axis_aligned_bbox(np.array(gt_bbox))
                    gt_bbox_ = [cx-(w-1)/2, cy-(h-1)/2, w, h]
                    target_pos, target_sz = np.array([cx, cy]), np.array([w, h])
                    state = SiamRPN_init(img, target_pos, target_sz, net)
                    pred_bbox = gt_bbox_
                    pred_bboxes.append(1)
                elif idx > frame_counter:
                    state = SiamRPN_track_upd(state, img, updatenet)
                    pred_bbox = cxy_wh_2_rect(state['target_pos'], state['target_sz'])
                    overlap = vot_overlap(pred_bbox, gt_bbox, (img.shape[1], img.shape[0]))
                    if overlap > 0:
                        # not lost
                        pred_bboxes.append(pred_bbox)
                    else:
                        # lost object
                        pred_bboxes.append(2)
                        frame_counter = idx + 5 # skip 5 frames
                        lost_number += 1
                else:
                    pred_bboxes.append(0)
                toc += cv2.getTickCount() - tic
                if idx == 0:
                    cv2.destroyAllWindows()
                if args.vis and idx > frame_counter:
                    cv2.polylines(img, [np.array(gt_bbox, np.int).reshape((-1, 1, 2))],
                            True, (0, 255, 0), 3)

                    pred_bbox = list(map(int, pred_bbox))
                    cv2.rectangle(img, (pred_bbox[0], pred_bbox[1]),
                                      (pred_bbox[0]+pred_bbox[2], pred_bbox[1]+pred_bbox[3]), (0, 255, 255), 3)
                    cv2.putText(img, str(idx), (40, 40), cv2.FONT_HERSHEY_SIMPLEX, 1, (0, 255, 255), 2)
                    cv2.putText(img, str(lost_number), (40, 80), cv2.FONT_HERSHEY_SIMPLEX, 1, (0, 0, 255), 2)

                    cv2.imshow(video.name, img)
                    cv2.waitKey(1)
            toc /= cv2.getTickFrequency()
            # save results
            video_path = os.path.join('results', args.dataset, model_name,
                    'baseline', video.name)
            if not os.path.isdir(video_path):
                os.makedirs(video_path)
            result_path = os.path.join(video_path, '{}_001.txt'.format(video.name))
            with open(result_path, 'w') as f:
                for x in pred_bboxes:
                    if isinstance(x, int):
                        f.write("{:d}\n".format(x))
                    else:
                        f.write(','.join([vot_float2str("%.4f", i) for i in x])+'\n')
            print('({:3d}) Video: {:12s} Time: {:4.1f}s Speed: {:3.1f}fps Lost: {:d}'.format(
                    v_idx+1, video.name, toc, idx / toc, lost_number))
            total_lost += lost_number
        print("{:s} total lost: {:d}".format(model_name, total_lost))

if __name__ == '__main__':
    main()

But I only get the following results:
![image](https://user-images.githubusercontent.com/49015333/87267095-575f8500-c4f9-11ea-921e-d7a7363b4773.png)
The above are the results you provided，
The following are my own evaluation results
Could you help me point out what went wrong with me？

**Comment by hds-xb** [comment]
I have the same problem, have you solved it? I test code on VOT2018, and I get 0.362(EAO).
